src/game/objects/pickup.c

```diff
@@ -104,3 +115,126 @@ void PickUpCollision(int16_t item_num, ITEM_INFO *lara_item, COLL_INFO *coll)
+                    }
+                    Overlay_AddPickup(item->object_number);
+                    Inv_AddItem(item->object_number);
+                    item->status = IS_INVISIBLE;
+                    RemoveDrawnItem(item_num);
+                    g_GameInfo.stats.pickup_count++;
+                    g_Lara.interact_target.is_moving = 0;
+                }
+            }
+        } else if (g_Lara.water_status == LWS_UNDERWATER) {
+            item->pos.x_rot = -25 * PHD_DEGREE;
+
+            if ((g_Input.action && lara_item->current_anim_state == AS_TREAD
+                 && g_Lara.gun_status == LGS_ARMLESS
+                 && !g_Lara.interact_target.is_moving)
+                || (g_Lara.interact_target.is_moving
+                    && g_Lara.interact_target.item_num == item_num)) {
+
+                if (TestLaraPosition(g_PickUpBoundsUW, item, lara_item)) {
+                    if (MoveLaraPosition(
+                            &g_PickUpPositionUW, item, lara_item)) {
+                        lara_item->anim_number = AA_PICKUP_UW;
+                        lara_item->current_anim_state = AS_PICKUP;
+
+                        lara_item->goal_anim_state = AS_TREAD;
+                        lara_item->frame_number =
+                            g_Anims[lara_item->anim_number].frame_base;
+                        g_Lara.interact_target.is_moving = 0;
+                        g_Lara.gun_status = LGS_HANDSBUSY;
+                    }
+                    g_Lara.interact_target.item_num = item_num;
+                } else if (
+                    g_Lara.interact_target.is_moving
+                    && g_Lara.interact_target.item_num == item_num) {
+                    g_Lara.interact_target.is_moving = 0;
+                    g_Lara.gun_status = LGS_ARMLESS;
+                }
+            } else if (
+                g_Lara.interact_target.item_num == item_num
+                && lara_item->current_anim_state == AS_PICKUP
+                && lara_item->frame_number
+                    == g_Anims[AA_PICKUP_UW].frame_base + 18) {
+                Overlay_AddPickup(item->object_number);
+                Inv_AddItem(item->object_number);
+                item->status = IS_INVISIBLE;
+                RemoveDrawnItem(item_num);
+                g_GameInfo.stats.pickup_count++;
+                return;
+            }
+
+            if (lara_item->current_anim_state == AS_PICKUP) {
+                if (lara_item->frame_number != AF_PICKUP_UW) {
+                    return;
+                }
+                Overlay_AddPickup(item->object_number);
+                Inv_AddItem(item->object_number);
+                item->status = IS_INVISIBLE;
+                RemoveDrawnItem(item_num);
+                g_GameInfo.stats.pickup_count++;
+                return;
+            }
+        }
+        item->pos.x_rot = rotx;
+        item->pos.y_rot = roty;
+        item->pos.z_rot = rotz;
```

Comment:
Why is it that we need to restore the item rotation here while `PickUpCollision` gets away with side effects?

Reply:
I'm not totally sure. This was added in TR5. All the items in TR1 were sprites, so it wasn't needed? Idk. I could ask TR5 people.

Reply:
From ChocolateFan:
```
lara yrot is copied to item yrot
and for safety xrot and zrot are copied too
this way the item is always aligned to lara
and this is the reason lara can pick them up from any direction
for other objects like switches
lara must have proper orientation
```

So I don't think TR1 has any consequences of not restoring the pickup's rotations, but it's probably good practice to do it.